Add Explicit K8s Compatibility Statement, Change EOM Date for Enterprise v0.25 (#310)
astronomer/docs: enterprise/version-compatibility-reference.md

The Astronomer Platform ships with and requires a number of adjacent technologies that support it, including Kubernetes, Helm and Apache Airflow itself. For users looking to install or upgrade Astronomer, we've provided a reference of all required tooling with corresponding versions that are compatible with each version of Astronomer Enterprise. For those running Astronomer Certified (our distribution of Apache Airflow) _without_ our platform, we've included a reference table below as well.

> In general, Astronomer Enterprise will support a version of Kubernetes (upstream and cloud-managed variants) through its End of Life. When it reaches End of Life, support for a given version of Kubernetes will be removed in the next major or minor release of Astronomer Enterprise.

Reviewer: ```suggestion
In general, Astronomer Enterprise supports a given version of Kubernetes (upstream and cloud-managed variants) through its End of Life. When that version of Kubernetes reaches its End of Life, support will be removed in the next major or minor release of Astronomer Enterprise.
```

Author: This was intentionally forward-looking to set expectation beyond current releases, hence 'will support' vs. 'supports'

Reviewer: Sounds good, just updated the suggestion!

> The Astronomer Platform ships with and requires a number of adjacent technologies that support it, including Kubernetes, Helm and Apache Airflow itself. For users looking to install or upgrade Astronomer, we've provided a reference of all required tooling with corresponding versions that are compatible with each version of Astronomer Enterprise. For those running Astronomer Certified (our distribution of Apache Airflow) _without_ our platform, we've included a reference table below as well.
> 
> In general, Astronomer Enterprise will support a given version of Kubernetes (upstream and cloud-managed variants) through its End of Life. When that version of Kubernetes reaches its End of Life, support will be removed in the next major or minor release of Astronomer Enterprise.
> 
> It's worth noting that while the tables below reference the minimum compatible versions, we typically recommend running the _latest_ of all tooling if possible.

Reviewer 2: ```suggestion
Astronomer Enterprise ships with and requires a number of adjacent technologies that support it, including Kubernetes, Helm and Apache Airflow itself. For users looking to install or upgrade Astronomer, we've provided a reference of all required tooling with corresponding versions and their compatibility status with Astronomer Enterprise. For those running [Astronomer Certified](image-architecture) outside of the context of our platform, we've included a reference table below as well.

It's worth noting that while the tables below reference the minimum compatible versions, we typically recommend running the _latest_ of all tooling if possible.
```

> ## Astronomer Enterprise

Reviewer 2: @jwitz I'd add an H3 here under this reference table titled - `### Kubernetes Version Support Policy`

A few tweaks here - 

> In general, Astronomer Enterprise will support a given version of Kubernetes through its End of Life. This includes Kubernetes upstream and cloud-managed variants like GKE, AKS, and EKS. When a version of Kubernetes reaches End of Life, support will be removed in the next major or minor release of Astronomer Enterprise.
>
> For more information on Kubernetes policies, refer to [Kubernetes Release History](https://kubernetes.io/releases/) or your cloud provider.

Reviewer: Shouldn't we include this in the release/ lifecycle policy doc instead of this doc? It's more of a policy than a compatibility consideration
